includes/wizards/newspack/class-custom-events-section.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Custom Events Section Object.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack\Wizards\Newspack;
+
+/**
+ * WordPress dependencies
+ */
+use WP_Error, WP_REST_Server;
+
+/**
+ * Internal dependencies
+ */
+use Newspack\Wizards\Wizard_Section;
+
+/**
+ * Custom Events Section Object.
+ *
+ * @package Newspack\Wizards\Newspack
+ */
+class Custom_Events_Section extends Wizard_Section {
+
+	/**
+	 * Register Wizard Section specific endpoints.
+	 *
+	 * @return void 
+	 */
+	public function register_rest_routes() {
+		register_rest_route(
+			NEWSPACK_API_NAMESPACE_V2,
+			'/wizard/analytics/ga4-credentials',
+			[
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => [ $this, 'api_set_ga4_credentials' ],
+				'permission_callback' => [ $this, 'api_permissions_check' ],
+				'args'                => [
+					'measurement_id'              => [
+						'sanitize_callback' => 'sanitize_text_field',
+						'validate_callback' => [ $this, 'validate_measurement_id' ],
+					],
+					'measurement_protocol_secret' => [
+						'sanitize_callback' => 'sanitize_text_field',
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Updates the GA4 crendetials
+	 *
+	 * @param WP_REST_Request $request The REST request.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function api_set_ga4_credentials( $request ) {
+		$measurement_id              = $request->get_param( 'measurement_id' );
+		$measurement_protocol_secret = $request->get_param( 'measurement_protocol_secret' );
+
+		if ( ! $measurement_protocol_secret ) {
+			return new WP_Error(
+				'newspack_analytics_wizard_invalid_params',
+				esc_html__( 'Invalid Measurement Protocol API Secret.', 'newspack' ),
+				[ 'status' => 400 ]
+			);
+		}
+
+		update_option( 'ga4_measurement_id', sanitize_text_field( $measurement_id ) );
+		update_option( 'ga4_measurement_protocol_secret', sanitize_text_field( $measurement_protocol_secret ) );
```

Comment:
I think these are being sanitized twice given the `sanitize_callback` specified in while registering the route.

Reply:
I agree, however, I am following [late escaping](https://docs.wpvip.com/security/validating-sanitizing-and-escaping/#h-always-escape-late) principles. FWIW - for the most part this code was [cloned](https://github.com/Automattic/newspack-plugin/blob/129db37d81d3f4b4eff10803e89aaed4c2804b08/includes/wizards/class-analytics-wizard.php#L106-L122). 

Let me know if you'd still like me to remove?

Reply:
Thanks for clarifying. I'm fine with the double sanitizing as it's not breaking anything 👍